src/client/components/common/Combobox.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  HStack,
+  Input,
+  ListItem,
+  InputProps,
+  VStack,
+  Text,
+  InputRightElement,
+  InputGroup,
+  IconButton,
+  Box,
+  forwardRef,
+} from '@chakra-ui/react'
+import { FixedSizeList, ListChildComponentProps } from 'react-window'
+import Downshift, {
+  DownshiftState,
+  GetItemPropsOptions,
+  StateChangeOptions,
+} from 'downshift'
+import { matchSorter, MatchSorterOptions } from 'match-sorter'
+import React, { FC, RefObject, useRef } from 'react'
+import { BiChevronDown, BiX } from 'react-icons/bi'
+import { FieldOption } from '../../../types/checker'
+
+enum DropdownDirection {
+  up,
+  down,
+}
+
+export interface ComboboxItem {
+  label: string
+  value: unknown
+}
+
+interface ItemRendererProps extends Omit<ListChildComponentProps, 'data'> {
+  data: {
+    items: ComboboxItem[]
+    highlightedIndex: number
+    selectedItem: FieldOption
+    topDropdownInset: number
+    getItemProps: (options: GetItemPropsOptions<ComboboxItem>) => unknown
+  }
+}
+
+/**
+ * Renders a single dropdown item.
+ *
+ * Used by react-window's `FixedSizeList` to efficiently render
+ * large lists. This helps reduce the performance impact of
+ * Downshift's event re-renders.
+ *
+ * @param props Props passed in by the `FixedSizeList` pure component.
+ *
+ * @returns A list item to be displayed in the dropdown
+ */
+const ItemRenderer: FC<ItemRendererProps> = (props) => {
+  const { items, getItemProps, highlightedIndex, topDropdownInset } = props.data
+  const item = items[props.index]
+
+  return (
+    <>
+      <ListItem
+        {...getItemProps({
+          style: {
+            ...props.style,
+            // purely for adding padding insets to FixedStyleList
+            // see https://codesandbox.io/s/react-window-list-padding-dg0pq
+            top: `${parseFloat(`${props.style.top}`) + topDropdownInset}px`,
```

Comment:
```suggestion
            top: `${parseFloat(`${props.style.top}`) + topDropdownInset}px`,
            listStyle: 'none',
```

Currently bullets are still shown for list items.

<img src='https://user-images.githubusercontent.com/3666479/116958543-9b608800-accd-11eb-8410-86a5da1526dd.png' width=500 />